analytics/pubxai/utils/utils.go

```diff
@@ -0,0 +1,612 @@
+	RenderedSize      string                 `json:"renderedSize"`
+}
+
+type AuctionBids struct {
+	AuctionDetail AuctionDetail          `json:"auctionDetail"`
+	FloorDetail   FloorDetail            `json:"floorDetail"`
+	PageDetail    PageDetail             `json:"pageDetail"`
+	DeviceDetail  DeviceDetail           `json:"deviceDetail"`
+	UserDetail    UserDetail             `json:"userDetail"`
+	ConsentDetail ConsentDetail          `json:"consentDetail"`
+	PmacDetail    map[string]interface{} `json:"pmacDetail"`
+	InitOptions   InitOptions            `json:"initOptions"`
+	Bids          []Bid                  `json:"bids"`
+	Source        string                 `json:"source"`
+}
+
+type WinningBid struct {
+	AuctionDetail AuctionDetail          `json:"auctionDetail"`
+	FloorDetail   FloorDetail            `json:"floorDetail"`
+	PageDetail    PageDetail             `json:"pageDetail"`
+	DeviceDetail  DeviceDetail           `json:"deviceDetail"`
+	UserDetail    UserDetail             `json:"userDetail"`
+	ConsentDetail ConsentDetail          `json:"consentDetail"`
+	PmacDetail    map[string]interface{} `json:"pmacDetail"`
+	InitOptions   InitOptions            `json:"initOptions"`
+	WinningBid    Bid                    `json:"winningBid"`
+	Source        string                 `json:"source"`
+}
+
+type PageDetail struct {
+	Host   string `json:"host"`
+	Path   string `json:"path"`
+	Search string `json:"search"`
+}
+
+type DeviceDetail struct {
+	Platform   string `json:"platform"`
+	DeviceType int    `json:"deviceType"`
+	DeviceOS   int    `json:"deviceOS"`
+	Browser    int    `json:"browser"`
+}
+
+type UserDetail struct {
+	UserIdTypes []string `json:"userIdTypes"`
+}
+
+type ConsentDetail struct {
+	ConsentTypes []string `json:"consentTypes"`
+}
+
+type InitOptions struct {
+	AuctionId    string `json:"auctionId"`
+	SamplingRate int    `json:"samplingRate"`
+	Pubxid       string `json:"pubxid"`
+}
+
+type FloorDetail struct {
+	FetchStatus     string `json:"fetchStatus"`
+	FloorProvider   string `json:"floorProvider"`
+	Location        string `json:"location"`
+	ModelVersion    string `json:"modelVersion"`
+	NoFloorSignaled bool   `json:"noFloorSignaled"`
+	SkipRate        int64  `json:"skipRate"`
+	Skipped         bool   `json:"skipped"`
+	SkippedReason   string `json:"skippedReason"`
+}
+
+type FloorData struct {
+	FetchStatus     string `json:"fetchStatus"`
+	FloorProvider   string `json:"floorProvider"`
+	Location        string `json:"location"`
+	ModelVersion    string `json:"modelVersion"`
+	NoFloorSignaled bool   `json:"noFloorSignaled"`
+	SkipRate        int64  `json:"skipRate"`
+	Skipped         bool   `json:"skipped"`
+	SkippedReason   string `json:"skippedReason"`
+}
+
+type AuctionDetail struct {
+	AdUnitCodes []string `json:"adUnitCodes"`
+	RefreshRank int64    `json:"refreshRank"`
+	AuctionId   string   `json:"auctionId"`
+	Timestamp   int64    `json:"timestamp"`
+}
+
+type UtilsService interface {
```

Comment:
Why do utils need to be a service?

Reply:
this was for mainly for the Unit tests, but anyway, since not using gomock anymore, utils is not a service anymore